### PR TITLE
Added support NX for SET command

### PR DESCRIPTION
**Pull request by chanducheryala**

### CODE REVIEW

File: tests/set_test.go [hunk 1]
```diff
@@ -21,3 +21,37 @@ func TestSet(t *testing.T) {
 		}
 	}
 }
+func TestSetWithNX(t *testing.T) {
+	conn := getLocalConnection()
+	for _, tcase := range []DTestCase{
+		{
+			InCmds: []string{"SET K V NX", "GET K"},
```
Comment (sanuj96) [inline tests/set_test.go:28]: Whetever keys you are planning to use in your test make sure to clean them up by calling deleteTestKeys utility

Comment (chanducheryala) [reply tests/set_test.go:28]: yeah, sure

Comment (sanuj96) [reply tests/set_test.go:28]: Add negative tests too

like SET a key, then SET the key again using NX param

**Review by sanuj96 — COMMENTED** [review]
Monor comments.

Please update your branch with latest master.

**Comment by chanducheryala** [comment]
@JyotinderSingh , can u please review the PR

**Comment by arpitbbhayani** [comment]
@chanducheryala Changes look good. Merging them. Thanks.

**Pull request merged**